File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        // 배포는 하루에 한번, 하루의 끝에 이어진다.
        int[] remains = new int[progresses.length];
        int [] days = new int [progresses.length];
        for (int i = 0; i< remains.length; i++) {
            remains[i] = 100 - progresses[i] ;
            if (remains[i] % speeds[i] == 0) { 
                days[i] = remains[i] / speeds[i];
            }else { 
                days[i] = (remains[i] / speeds[i]) +1;
            }
        }
        
        ArrayList <Integer>  arr = new ArrayList <>();
        
        int prev=  days[0];
        int cnt = 1;

        for (int i = 1; i < days.length; i++ )  {
            if (prev < days [i]) {
                arr.add (cnt);
                cnt =1;
                prev = days[i];
               
            }
            
            else if  (prev == days[i]) {
                cnt +=1;
            }
            else { 
                cnt +=1;
            }
        }
        
        arr.add(cnt);
        int [] answer = new int [arr.size()];
        int idx = 0;
        for (int num : arr) {
            answer [idx] = num;
            idx +=1;
        }
      
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    
    
    # 작업 남은거
    
    remains = []
    
    for i in range (0, len(progresses)) :
        remains.append( ((100- progresses[i]) // speeds[i], (100-progresses[i]) % speeds[i]))
    
      
    new_li = []
    for tup in remains :
        day = tup[0]
        remain = tup[1]
        if remain != 0 :
            day = day +1
            remain = 0
        
        new_li.append((day,remain))
            
  
    print("new", new_li)
    prev = new_li[0][0]
    cnt = 1
    for i in range(1, len(new_li)) :
        if prev >= new_li[i][0] :
            cnt +=1
        else :
            answer.append(cnt)
            cnt = 1
            prev = new_li[i][0]
            
    answer.append(cnt)
     
        
            
        
        
    
    
    return answer